Wrappers/Python/cil/processors/FluxNormaliser.py

```diff
@@ -0,0 +1,458 @@
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+# Authors:
+# CIL Developers, listed at: https://github.com/TomographicImaging/CIL/blob/master/NOTICE.txt
+
+from cil.framework import Processor, AcquisitionData
+from cil.utilities import multiprocessing as cil_mp
+
+import numpy
+import logging
+import matplotlib.pyplot as plt
+import numba
+
+log = logging.getLogger(__name__)
+
+class FluxNormaliser(Processor):
+    r'''
+    Flux normalisation based on float or region of interest
+
+    This processor reads in an AcquisitionData and normalises it by flux from
+    a float or array of float values, or the mean flux in a region of interest.
+
+    Parameters:
+    -----------
+    flux: float or list of floats, optional
+        The value to divide the image by. If flux is a list it must have length 
+        equal to the number of projections in the dataset. If flux=None, calculate
+        flux from the roi.
+    
+    roi: dict, optional
+        Dictionary describing the region of interest containing the background
+        in the image. The roi is specified as `{'horizontal':(start,stop), 
+        'vertical':(start,stop)}`. If an axis is not specified in the roi dictionary, 
+        the full range will be used.
+
+    target: {'mean', 'first'} or float, default='mean'
+        The value to scale the normalised data with. If float, the data is scaled 
+        to the float value. If string 'mean' the data is scaled to the mean value 
+        of the input flux or flux across all rois. If 'first' the data is scaled to 
+        the first input flux value or the flux in the roi of the first projection.
+        Default is 'mean'
+
+    Returns:
+    --------
+    Output: AcquisitionData normalised by flux
+
+    Example
+    -------
+    >>> from cil.processors import FluxNormaliser
+    >>> processor = FluxNormaliser(flux=10)
+    >>> processor.set_input(data)
+    >>> data_norm = processor.get_output()
+
+    Example
+    -------
+    >>> from cil.processors import FluxNormaliser
+    >>> processor = FluxNormaliser(flux=np.arange(1,2,(2-1)/(data.get_dimension_size('angle'))))
+    >>> processor.set_input(data)
+    >>> data_norm = processor.get_output()
+
+    Example
+    -------
+    >>> from cil.processors import FluxNormaliser
+    >>> processor = FluxNormaliser(roi={'horizontal':(5, 15)})
+    >>> processor.set_input(data)
+    >>> data_norm = processor.get_output()
+
+    Note
+    ----
+    The roi indices provided are start inclusive, stop exclusive.
+    All elements along a dimension will be included if the axis does not appear 
+    in the roi dictionary
+    '''
+
+    def __init__(self, flux=None, roi=None, target='mean', accelerated=True):
```

Comment:
maybe for a discussion, but do we need to expose `accelerated` here? As numba is a dependency I'd be happy to use it without giving an alternative route through the code.